src/stackset.ts

```diff
@@ -658,6 +658,10 @@ export class StackSet extends Resource implements IStackSet {
       templateUrl: props.template.templateUrl,
       stackInstancesGroup: Lazy.any({ produce: () => { return this.stackInstances; } }),
     });
+
+    // the file asset bucket deployment needs to complete before the stackset can deploy
+    const fileAssetResource = scope.node.tryFindChild(fileAssetResourceName);
+    fileAssetResource && stackSet.node.addDependency(fileAssetResource);
```

Comment:
i was thinking about forcing a dependency on all sibling resources to the StackSet resource here. but opted for this more targeted approach. but i still like the idea of forcing the StackSet resource to always be last in the deployment if there is a clean way to do that.